sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala

```diff
@@ -661,14 +680,32 @@ private object CosmosPartitionPlanner extends BasicLoggingTrait {
           .fromArray(feedRanges)
           .flatMap(
             normalizedRange =>
-              PartitionMetadataCache.apply(
-                userConfig,
-                cosmosClientConfig,
-                cosmosClientStateHandle,
-                cosmosContainerConfig,
-                normalizedRange,
-                maxStaleness
-            ))
+              if (isChangeFeed != partitionConfig.partitioningStrategy != PartitioningStrategies.Restrictive) {
```

Comment:
!isChangeFeed ?

Reply:
Fixed the typo - good catch. Also added a comment explaining the logic